nullaway/src/test/java/com/uber/nullaway/NullAwayContractsBooleanTests.java

```diff
@@ -0,0 +1,351 @@
+package com.uber.nullaway;
+
+import com.google.errorprone.CompilationTestHelper;
+import java.util.Arrays;
+import org.junit.Test;
+
+public class NullAwayContractsBooleanTests extends NullAwayTestsBase {
```

Comment:
These tests look great!

Reply:
Overall, this is an _impressive_ test suite. My only worry here is that all tests are on APs that are also arguments or local variables. I wonder how well the logic works (particularly the parts that check for `Nullness.NULL` rather than `Nullness.NULLABLE`) with other tracked expressions / access-paths. Could we have just a few cases calling "assumed getters" (no arg methods) or `Map::get` with constant keys?

Reply:
Good idea, I've updated a few cases with otherwise equivalent code using `Map::get` and an assumed-getter